music21/romanText/writeRoman.py

```diff
@@ -0,0 +1,481 @@
+# -*- coding: utf-8 -*-
+# ------------------------------------------------------------------------------
+# Name:         romanText/writeRoman.py
+# Purpose:      Writer for the 'RomanText' format
+#
+# Authors:      Mark Gotham
+#
+# Copyright:    Copyright © 2020 Michael Scott Cuthbert and the music21 Project
+# License:      BSD, see license.txt
+# ------------------------------------------------------------------------------
+'''
+Writer for the 'RomanText' format (Tymoczko, Gotham, Cuthbert, & Ariza ISMIR 2019)
+'''
+
+import fractions
+import unittest
+
+from typing import Optional, Union
+
+from music21 import base
+from music21 import metadata
+from music21 import meter
+from music21 import prebase
+from music21 import roman
+from music21 import romanText
+from music21 import stream
+
+
+# ------------------------------------------------------------------------------
+
+class RnWriter(prebase.ProtoM21Object):
+    '''
+    Extracts the relevant information from a stream of Roman numeral objects for
+    writing to text files in the 'RomanText' format (rntxt).
+    Writing rntxt is handled externally in the subConverters module so
+    most users will never need to call this class directly, and will only invoke
+    it indirectly through .write('rntxt').
+    Possible exceptions include users who want to convert Roman numberals into rntxt type
+    strings and want to work with those strings directly, without writing to disc.
```

Comment:
Not to be too American about it, but I think for computers it's always dis**k**.